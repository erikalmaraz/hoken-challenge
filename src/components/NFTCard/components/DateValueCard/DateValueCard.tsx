import React from 'react'
import { StyledDateValueCard } from './date-value-card.styles'
import { DateValueCardProps } from './date-value-card.interfaces'
const DateValueCard = ({
  currentValue,
  daysLeft,
  currency,
  ...props
}: DateValueCardProps): JSX.Element => {
  return (
    <StyledDateValueCard>
      <div className="footer-card--amount">
        <img src="./assets/images/icon-ethereum.svg" alt="Etherium icon" />
        <span data-testid="footer-card--value">
          {currentValue} {currency}
        </span>
      </div>
      <div className="footer-card--time">
        <img src="./assets/images/icon-clock.svg" alt="Clock icon" />
        <span data-testid="footer-card--days">{daysLeft} days left</span>
      </div>
    </StyledDateValueCard>
  )
}

DateValueCard.displayName = 'DateValueCard'

export default DateValueCard
