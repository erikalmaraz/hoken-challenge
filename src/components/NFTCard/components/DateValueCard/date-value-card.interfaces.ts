export interface DateValueCardProps {
  currentValue: number
  daysLeft: number
  currency: string
}
