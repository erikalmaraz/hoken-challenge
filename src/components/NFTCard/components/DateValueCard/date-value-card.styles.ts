import styled from 'styled-components'
import { primary } from '../../../../styles/colors'

export const StyledDateValueCard = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-top: 1.2em;
  .footer-card--amount {
    display: flex;
    align-items: center;
    img {
      width: 13px;
      margin-right: 0.5em;
    }
    span {
      font-weight: bold;
      font-size: 1.03em;
      color: ${primary.cyan};
    }
  }
  .footer-card--time {
    display: flex;
    align-items: center;
    img {
      width: 21px;
      margin-right: 0.5em;
    }
    span {
      font-size: 1.03em;
      font-weight: 300;
    }
  }
`
