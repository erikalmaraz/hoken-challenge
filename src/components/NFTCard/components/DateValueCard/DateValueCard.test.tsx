import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import DateValueCard from './DateValueCard'

it('Render DateValueCard component', () => {
  const NFTInfo = {
    days: 3,
    currency: 'ETH',
    value: 0.041
  }
  const component = render(
    <DateValueCard
      currentValue={NFTInfo.value}
      currency={NFTInfo.currency}
      daysLeft={NFTInfo.days}
    />
  )
  expect(component).toMatchSnapshot()
})
