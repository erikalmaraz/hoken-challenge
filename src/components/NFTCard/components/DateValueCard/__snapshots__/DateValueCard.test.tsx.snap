// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DateValueCard component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ fUzKTP"
      >
        <div
          class="footer-card--amount"
        >
          <img
            alt="Etherium icon"
            src="./assets/images/icon-ethereum.svg"
          />
          <span
            data-testid="footer-card--value"
          >
            0.041
             
            ETH
          </span>
        </div>
        <div
          class="footer-card--time"
        >
          <img
            alt="Clock icon"
            src="./assets/images/icon-clock.svg"
          />
          <span
            data-testid="footer-card--days"
          >
            3
             days left
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ fUzKTP"
    >
      <div
        class="footer-card--amount"
      >
        <img
          alt="Etherium icon"
          src="./assets/images/icon-ethereum.svg"
        />
        <span
          data-testid="footer-card--value"
        >
          0.041
           
          ETH
        </span>
      </div>
      <div
        class="footer-card--time"
      >
        <img
          alt="Clock icon"
          src="./assets/images/icon-clock.svg"
        />
        <span
          data-testid="footer-card--days"
        >
          3
           days left
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
